### ISSUE TITLE
Modifiers do not show in JEI if the first world you load into is a server

### ISSUE THREAD

**Issue by ChiefArug** [SlimeKnights/TinkersConstruct#5074]
### Minecraft Version

1.18.2

### Forge Version

40.2.0

### Mantle Version

3.6.2.92

### Tinkers' Construct Version

1.9.43

### Describe your issue

If you join a server before joining a singleplayer world then modifiers in the ingredient list do not show up (I also had it happen with casting recipes, but haven't been able to reproduce that).
This error appears in the logs when joining: 
```
 Caught an error from mod plugin: class slimeknights.tconstruct.plugin.jei.JEIPlugin tconstruct:jei_plugin
java.lang.IllegalArgumentException: ingredients must not be empty.
```

If I then join a single player world, the modifiers still do not show up, but the error does not happen again.

However, if I start the game and join a single player world before a server, then everything works as expected, even if I later join a server.

### Crash Report

https://gist.github.com/ChiefArug/84aeace93940499afee9513eac066f77

### Other mods

Small pack of ~30 mods which shouldn't be affecting Tinkers (and are not mentioned in the stacktrace)
Full mod list is a part of the logs gist if needed.

### Tried reproducing with just Tinkers?

No

### Performance Enchancers

None of the above

### Searched for known issues?

Checked pinned issues, Searched open issues, Searched closed issues, Checked the FAQ

**Comment by FoxMcloud5655** [comment]
I think this is specifically related to JEI 10+ breaking something, as we're seeing this happen in Draconic Evolution 3 for 1.18.2 and also already fixed it.  May or may not be related, but worth a mention anyways.

**Comment by KnightMiner** [comment]
The specific thing that broke is new, one of our caches loading too early due to vanilla logic.

**Comment by KnightMiner** [comment]
Fixed in 3.6.3.111.
